apis/v1beta1/types.go

```diff
@@ -211,6 +218,51 @@ type VirtualMachineCloneSpec struct {
 	DataDisks []VSphereDisk `json:"dataDisks,omitempty"`
 }
 
+// VirtualMachineResources is the definition of the VM's cpu and memory
+// reservations, limits and shares.
+// +kubebuilder:validation:MinProperties=1
+type VirtualMachineResources struct {
+	// requests is the definition of the VM's cpu (in hertz, rounded up to the nearest MHz)
+	// and memory (in bytes, rounded up to the nearest MiB) reservations
+	// +optional
+	Requests VirtualMachineResourceSpec `json:"requests,omitempty,omitzero"`
+	// limits is the definition of the VM's cpu (in hertz, rounded up to the nearest MHz)
+	// and memory (in bytes, rounded up to the nearest MiB) limits
+	// +optional
+	Limits VirtualMachineResourceSpec `json:"limits,omitempty,omitzero"`
+	// shares is the definition of the VM's cpu and memory shares
+	// +optional
+	Shares VirtualMachineResourceShares `json:"shares,omitempty,omitzero"`
+}
+
+// VirtualMachineResourceSpec is the numerical definition of memory and cpu quantity for the
+// given VM hardware policy.
+// +kubebuilder:validation:MinProperties=1
+type VirtualMachineResourceSpec struct {
+	// cpu is the definition of the cpu quantity for the given VM hardware policy
+	// +optional
+	Cpu resource.Quantity `json:"cpu,omitempty"` //nolint:revive
```

Comment:
Sorry might have missed a previous discussion. Is there a reason why we are ignoring the finding instead of using CPU? (same below)

Ah maybe to align to vm-operator. Please drop the nolint and use CPU instead. I prefer getting this right over alignment to vm-operator :)